### PR TITLE
Split Test class ItMultiDomainModelsWithLoadBalancer

### PR DESCRIPTION
**Pull request by xiancao**
Split Test class ItMultiDomainModelsWithLoadBalancer to make tests more stable.

Jenkins result:
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/13525/
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/13595/ (the release/3.4 nightly also has these two tests failed)

### CODE REVIEW

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiNegative.java [hunk 1]
```diff
@@ -0,0 +1,220 @@
+// Copyright (c) 2022, Oracle and/or its affiliates.
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiNegative.java:1]: Can we move this usecase to a existing MiiDomain class

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiRollingRestartLivenessProbe.java [hunk 2]
```diff
@@ -0,0 +1,618 @@
...
+      logger.info("NGINX service name: {0}", nginxServiceName);
+      nodeportshttp = getServiceNodePort(nginxNamespace, nginxServiceName, "http");
+      logger.info("NGINX http node port: {0}", nodeportshttp);
+    }
+  }
+
+  /**
+   * Verify liveness probe by killing managed server process 3 times to kick pod container auto-restart.
+   */
+  @Test
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiRollingRestartLivenessProbe.java:177]: Can u try to merge the test method into ItLivenessProbeCustomization.java to aggregate all livenessProbe scenario.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiRollingRestartLivenessProbe.java [hunk 3]
```diff
@@ -0,0 +1,618 @@
...
+  /**
+   * Test rolling restart for a multi-clusters domain.
+   * Make sure pods are restarted only once.
+   * Verify all pods are terminated and restarted only once
+   * Rolling restart triggered by changing:
+   * imagePullPolicy: IfNotPresent --> imagePullPolicy: Never
+   * Verify domain changed event is logged.
+   */
+  @Test
+  @DisplayName("Verify server pods are restarted only once by changing the imagePullPolicy in multi-cluster domain")
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiRollingRestartLivenessProbe.java:279]: Move the tests into a java class where we verify the  server pod restart on domain attribute change. e.g.  ItPodsRestart

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMultiDomainModels.java [hunk 4]
```diff
@@ -130,7 +130,6 @@ void testScaleClustersAndAdminConsoleLogin(String domainType) {
     // In OKD cluster, we need to get the routeHost for the external admin service
     String routeHost = createRouteForOKD(getExternalServicePodName(adminServerPodName), domainNamespace);
 
-
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMultiDomainModels.java:133]: Do we not verify these usecase scale and access the console on three type of domains in other classes.  If so why we need this class

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMultiDomainModelsScale.java [hunk 5]
```diff
@@ -141,7 +112,6 @@ class ItMultiDomainModelsWithLoadBalancer {
   private static final String WLDF_OPENSESSION_APP = "opensessionapp";
   private static final String WLDF_OPENSESSION_APP_CONTEXT_ROOT = "opensession";
   private static final String wlSecretName = "weblogic-credentials";
-  private static final String DATA_HOME_OVERRIDE = "/u01/mydata";
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMultiDomainModelsScale.java:144]: There is no rolling restart happening on the class as mentioned on class level javadoc  "rolling restart behavior in a multi-cluster MII domain "

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiRollingRestartLivenessProbe.java (hunk 3 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiRollingRestartLivenessProbe.java:279]: There is a method testServerPodsRestartByChangingImagePullPolicy() in ItPodsRestart which tests the same usecase. I will delete this method.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMultiDomainModels.java (hunk 4 above)
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMultiDomainModels.java:133]: If this is for gating test, can we add a Auxiliary Image model.

**Review by anpanigr — COMMENTED** [review]
Modify the kindtest.sh  and pom.xml file to accommodate the class file name change.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiNegative.java (hunk 1 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiNegative.java]: moved it to ItMiiDomain.java

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiRollingRestartLivenessProbe.java (hunk 2 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiRollingRestartLivenessProbe.java]: moved it to ItLivenessProbeCustomization.java

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMultiDomainModels.java (hunk 4 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMultiDomainModels.java:133]: done

**Review by anpanigr — APPROVED** [review]

**Review by sankarpn — APPROVED** [review]
lgtm

**Pull request merged**